scripts/py_matter_yamltests/matter_yamltests/constraints.py

```diff
@@ -502,14 +508,20 @@ def get_reason(self, value, value_type_name) -> str:
             if value >= -9223372036854775807 and value <= 9223372036854775807:
                 types.append('nullable_int64s')
 
+            if (value >= -3.402823466E+38 and value <= -1.175494351E-38) or value == 0.0 or (value >= 1.175494351E-38 and value <= 3.402823466E+38):
```

Comment:
Wonder whether it's worth factoring this complicated condition used in multiple places out into a helper function....  Perhaps isFloatInBounds(x, y) that tests for "x <= num <= y or num == 0 or -y <= num <= -x" or some such.